### PR TITLE
Authenticate

### PR DESCRIPTION
**Pull request by 0xFirekeeper**
## PR-Codex overview
This PR enhances wallet authentication by adding a new method. It also introduces a utility function for generating a payload to sign. 

### Detailed summary
- Added `Authenticate` method to `IThirdwebWallet` interface and `PrivateKeyWallet` and `SmartWallet` classes
- Introduced `GenerateSIWE` utility function in `Utils.cs`
- Added `LoginPayload` and `LoginPayloadData` classes
- Included tests for `GenerateSIWE` function with different scenarios

> The following files were skipped due to too many changes: `Thirdweb.Tests/Thirdweb.Utils.Tests.cs`, `Thirdweb.Console/Program.cs`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: Thirdweb/Thirdweb.Wallets/IThirdwebWallet.cs [hunk 1]
```diff
@@ -16,11 +17,64 @@ public interface IThirdwebWallet
...
+
+        [JsonProperty("invalid_before", NullValueHandling = NullValueHandling.Ignore)]
+        public string InvalidBefore { get; set; }
+
+        [JsonProperty("resources", NullValueHandling = NullValueHandling.Ignore)]
+        public List<string> Resources { get; set; }
+
+        public LoginPayloadData()
+        {
+            Type = "evm";
```
Comment (jnsdls) [inline Thirdweb/Thirdweb.Wallets/IThirdwebWallet.cs:77]: Don't need this anymore, we used to use it when we also supported Solana but it's not actually part of any spec

Comment (0xFirekeeper) [reply Thirdweb/Thirdweb.Wallets/IThirdwebWallet.cs:77]: I think we still use it in verify js code no?

Comment (0xFirekeeper) [reply Thirdweb/Thirdweb.Wallets/IThirdwebWallet.cs:77]: or at least check for it, I remember last week I glanced over some code that did the check 🤔

Comment (0xFirekeeper) [reply Thirdweb/Thirdweb.Wallets/IThirdwebWallet.cs:77]: well in this case we don't use it anyway so would only be a problem if someone hits old sdk with this payload type and it expects type, will remove

Comment (0xFirekeeper) [reply Thirdweb/Thirdweb.Wallets/IThirdwebWallet.cs:77]: nvm it stopped working on local server without it, keeping

**Pull request merged**